### PR TITLE
feat: shaping out the lockbox contract

### PR DESCRIPTION
**Pull request by kupermind**
- Adding necessary functions to do the e2e lockbox workflow;
- Adding necessary accounts for the forked testing.

### CODE REVIEW

File: solidity/nft_token.sol [hunk 1]
```diff
@@ -10,42 +10,43 @@ struct Position {
...
-    address public whirlpool;
+    address public constant orca = address"whirLbMiicVdio4qvUfM5KAg6Ct8VwpYzGff3uctyCc";
+    address public pool;
     address public pdaProgram;
-    address public mintErc20;
-    address public pdaERC20Account;
-    // TODO: Change for the latter one in production
-    uint64 public pdaHeader = 0x55bdfe33;//0xd0f7407ae48fbcaa;
-    // PDA seed
+    address public bridgedTokenMint;
```
Comment (mariapiamo) [inline solidity/nft_token.sol:18]: why we are using the word "bridged" here?

File: solidity/nft_token.sol [hunk 2]
```diff
@@ -10,42 +10,43 @@ struct Position {
...
+    // Program PDA bump
     bytes1 public pdaBump;
     int32 public constant min_tick_lower_index = -443632;
     int32 public constant max_tick_lower_index = 443632;
-    //address constant tokenProgram = address"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA";
 
-    uint64 public numTokenAccounts;
+    // Total number of token accounts (even those that hold no positions anymore)
+    uint64 public numPositionAccounts;
+    // First available account index in the set of accounts;
```
Comment (mariapiamo) [inline solidity/nft_token.sol:31]: why first and not last?

File: solidity/nft_token.sol [hunk 3]
```diff
@@ -59,190 +60,245 @@ contract nft_token {
...
+        uint64 positionLiquidity = mapPositionAccountLiquidity[positionAddress];
         // Check that the token account exists
-        if (mapTokenAccountLiquidity[tx.accounts.pdaTokenAccount.key] == 0) {
+        if (positionLiquidity == 0) {
             revert("No liquidity on a provided token account");
         }
 
-        // Transfer ERC20 tokens to the pdaERC20Account address of this program
+        // Check the requested amount to be smaller or equal than the position liquidity
+        if (amount > positionLiquidity) {
```
Comment (mariapiamo) [inline solidity/nft_token.sol:183]: I had in mind that in this case we considered the second-to-last nft, and so on. It is likely the the protocol will have an amount larger than any liquidity position

File: solidity/nft_token.sol (hunk 1 above)
Comment (kupermind) [reply solidity/nft_token.sol:18]: This is the mint account of the bridged token - the token that will be bridged.

File: solidity/nft_token.sol (hunk 2 above)
Comment (kupermind) [reply solidity/nft_token.sol:31]: The first available one (the oldest one).

File: solidity/nft_token.sol (hunk 3 above)
Comment (kupermind) [reply solidity/nft_token.sol:183]: We can only go one NFT at a time.

**Pull request merged**